File: Android/app/src/main/java/com/edu/primary/database/dao/ChatMessageDao.java
package com.edu.primary.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.edu.primary.database.entity.ChatMessageEntity;

import java.util.List;

@Dao
public interface ChatMessageDao {
    @Insert
    long insertMessage(ChatMessageEntity message);

    @Query("SELECT * FROM chat_messages WHERE userId = :userId ORDER BY timestamp ASC")
    List<ChatMessageEntity> getMessagesByUser(long userId);

    @Query("DELETE FROM chat_messages WHERE userId = :userId")
    void clearMessages(long userId);

    @Query("SELECT COUNT(*) FROM chat_messages WHERE userId = :userId")
    int getMessageCount(long userId);
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/dao/ChatMessageDao.kt
package com.edu.primary.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.edu.primary.data.local.entity.ChatMessageEntity

@Dao
interface ChatMessageDao {
    @Insert
    suspend fun insert(message: ChatMessageEntity): Long

    @Query("SELECT * FROM chat_messages WHERE userId = :userId ORDER BY timestamp ASC")
    suspend fun getByUser(userId: Long): List<ChatMessageEntity>

    @Query("DELETE FROM chat_messages WHERE userId = :userId")
    suspend fun clearByUser(userId: Long)

    @Query("SELECT COUNT(*) FROM chat_messages WHERE userId = :userId")
    suspend fun getCount(userId: Long): Int
}
